src/exoticatechnologies/modifications/exotics/impl/GuardianShield.kt

```diff
@@ -261,54 +349,72 @@ class GuardianShield(key: String, settings: JSONObject) : Exotic(key, settings)
             log("<-- removeDrone()\tdrone: ${drone}\tsystemDeactivatedForGood: ${systemDeactivatedForGood.get()}")
         }
 
+        @Synchronized
         override fun spawnDrone(): ShipAPI {
-
-            val drone: ShipAPI = if(SPAWN_SHIELD_DRONE_USING_FXDRONE) {
-                val spec = Global.getSettings().getHullSpec(getDroneHullId())
-                val v = Global.getSettings().createEmptyVariant(getDroneVariant(), spec)
-                val fxDrone: ShipAPI = Global.getCombatEngine().createFXDrone(v)
-                fxDrone.layer = CombatEngineLayers.ABOVE_SHIPS_AND_MISSILES_LAYER
-                fxDrone.owner = ship.originalOwner
-                fxDrone.mutableStats.hullDamageTakenMult.modifyMult(INVULNERABLE_SHIELD_DRONE, 0f) // so it's non-targetable
-                fxDrone.isDrone = true
-                fxDrone.aiFlags.setFlag(ShipwideAIFlags.AIFlags.DRONE_MOTHERSHIP, 100000f, ship)
-                fxDrone.collisionClass = CollisionClass.FIGHTER
-                fxDrone.shipAI = null
-                fxDrone.isForceHideFFOverlay = true
-                fxDrone.setRenderBounds(false)
-                fxDrone.giveCommand(ShipCommand.SELECT_GROUP, null, 0)
-                Global.getCombatEngine().addEntity(fxDrone)
-
-                fxDrone
+            log("--> spawnDrone()\tdrone = ${drone}\tdrone is null ? ${drone == null}\townerShip = ${ship}")
+            // Since I have noticed multiple GuardianShields appear on just one multimodule ship,
+            // which had only one instance of GuardianShield in it's main module and nowhere else,
+            // this leads me to suspect that there is either a race-condition (or multiple threads) executing this somehow
+            // so lets go with the doubly-locked 'singleton' pattern to ensure only and exactly one drone is spawned
+            // when this method is called multiple times
```

Comment:
get rid of this because the problem was rooted in the now-refactored `shouldAffectModules()` method which automatically shared it to all other ship modules implicitly.

this is essentially just my "what if..." babbling before i did the exploratory work and found out the real culprit and fixed it with the new `shouldShareEffectsToOtherModules()` method

Reply:
done. resolving.